src/cplscheme/CouplingData.cpp

```diff
@@ -73,26 +93,28 @@ int CouplingData::meshDimensions() const
 
 void CouplingData::storeIteration()
 {
-  _previousIteration = this->values();
-  if (this->hasGradient()) {
-    PRECICE_ASSERT(this->gradientValues().size() > 0);
-    _previousIterationGradients = this->gradientValues();
-  }
+  const auto stamples = this->getStamples();
+  PRECICE_ASSERT(stamples.size() > 0);
+  this->sample()     = stamples.back().sample;
+  _previousIteration = this->sample();
 }
 
-const Eigen::VectorXd CouplingData::previousIteration() const
+const Eigen::VectorXd CouplingData::previousIteration()
 {
-  return _previousIteration;
+  if (_previousIteration.values.size() == 0) { // @todo work-around for direct mesh access, if _previousIteration was initialized with incorrect size.
+    _previousIteration = time::Sample{Eigen::VectorXd::Zero(getSize())};
```

Comment:
Would we still need this if we allocate in `initialize`? (see comments on `SolverInterfaceImpl` below)

Reply:
Good point. Seems to work now :) 95bde9138a9a8e15a517c019a86b84e440e3dc3d